presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsAnnotationProcessor.java

```diff
@@ -0,0 +1,243 @@
+                .setHighValue(maxValue)
+                .setNullsFraction(nullFraction)
+                .setAverageRowSize(firstFiniteValue(scalarStatsHeader.getAvgRowSize(), averageRowSize, returnNaNIfTypeWidthUnknown(getReturnTypeWidth(callExpression, UNKNOWN))))
+                .setDistinctValuesCount(processDistinctValuesCount(outputRowCount, nullFraction, scalarStatsHeader.getDistinctValuesCount(), distinctValuesCount)).build();
+    }
+
+    private static double processDistinctValuesCount(double outputRowCount, double nullFraction, double distinctValuesCountFromConstant, double distinctValuesCount)
+    {
+        if (isFinite(distinctValuesCountFromConstant)) {
+            if (nearlyEqual(distinctValuesCountFromConstant, NON_NULL_ROW_COUNT.getValue(), 0.1)) {
+                distinctValuesCountFromConstant = outputRowCount * (1 - firstFiniteValue(nullFraction, 0.0));
+            }
+            else if (nearlyEqual(distinctValuesCount, ROW_COUNT.getValue(), 0.1)) {
+                distinctValuesCountFromConstant = outputRowCount;
+            }
+        }
+        double distinctValuesCountFinal = firstFiniteValue(distinctValuesCountFromConstant, distinctValuesCount);
+        if (distinctValuesCountFinal > outputRowCount) {
+            distinctValuesCountFinal = NaN;
+        }
+        return distinctValuesCountFinal;
+    }
+
+    private static double processSingleArgumentStatistic(
+            double outputRowCount,
+            double nullFraction,
+            CallExpression callExpression,
+            List<Double> sourceStats,
+            int sourceStatsArgumentIndex,
+            StatsPropagationBehavior operation)
+    {
+        // sourceStatsArgumentIndex is index of the argument on which
+        // ScalarPropagateSourceStats annotation was applied.
+        double statValue = NaN;
+        if (operation.isMultiArgumentStat()) {
+            for (int i = 0; i < sourceStats.size(); i++) {
+                if (i == 0 && operation.isSourceStatsDependentStats() && isFinite(sourceStats.get(i))) {
+                    statValue = sourceStats.get(i);
+                }
+                else {
+                    switch (operation) {
+                        case MAX_TYPE_WIDTH_VARCHAR:
+                            statValue = returnNaNIfTypeWidthUnknown(getTypeWidthVarchar(callExpression.getArguments().get(i).getType()));
+                            break;
+                        case USE_MIN_ARGUMENT:
+                            statValue = min(statValue, sourceStats.get(i));
+                            break;
+                        case USE_MAX_ARGUMENT:
+                            statValue = max(statValue, sourceStats.get(i));
+                            break;
+                        case SUM_ARGUMENTS:
+                            statValue = statValue + sourceStats.get(i);
+                            break;
+                        case SUM_ARGUMENTS_UPPER_BOUNDED_TO_ROW_COUNT:
```

Comment:
Sorry if I've missed discussion on this in the past but what are the cases where we want the behavior of `SUM_ARGUMENTS` as opposed to `SUM_ARGUMENTS_UPPER_BOUNDED_TO_ROW_COUNT`

Reply:
Useful when a function developer want to use SUM of arguments stats for computing distinctValuesCount, which can sometimes exceed the row count and may not be the intention of the developer.

The discussion was, previously we were upper bounding by default which is not so helpful. In this way, for computing distinct values count through SUM_ARGUMENTS, user has better control, whether or not he wants to upper bound to row count. May be you or @aaneja has a better perspective on this.
Link to the discussion: https://github.com/prestodb/presto/pull/23545#discussion_r1778295120